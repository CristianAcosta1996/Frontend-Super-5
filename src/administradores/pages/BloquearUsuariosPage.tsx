import { Box, IconButton, Tooltip, Typography } from "@mui/material";
import {
  DataGrid,
  GridColDef,
  GridRenderCellParams,
  GridTreeNodeWithRender,
} from "@mui/x-data-grid";
import {
  useBloquearUsuarioMutation,
  useEliminarUsuarioMutation,
  useGetUsuariosQuery,
} from "../../store/super5/super5Api";
import { Person, PersonOff, PersonRemove } from "@mui/icons-material";

export const BloquearUsuariosPage = () => {
  const { data, isLoading } = useGetUsuariosQuery();
  return (
    <Box className="animate__animated animate__fadeIn" sx={{ height: "98vh" }}>
      <DataGrid
        columns={columns}
        loading={isLoading}
        rows={data || []}
        slots={{
          noRowsOverlay: () => (
            <Box
              height="100%"
              display="flex"
              alignItems="center"
              justifyContent="center"
            >
              <Typography textAlign="center" variant="h6" component="h6">
                No hay datos
              </Typography>
            </Box>
          ),
        }}
      />
    </Box>
  );
};

const columns: GridColDef[] = [
  {
    field: "id",
    headerName: "ID",
    width: 80,
  },
  { field: "usuario", headerName: "Usuario", width: 100 },
  { field: "nombre", headerName: "Nombre", width: 100 },
  { field: "apellido", headerName: "Apellido", width: 100 },
  { field: "eliminado", headerName: "Eliminado", width: 90, type: "boolean" },
  { field: "bloqueado", headerName: "Bloqueado", width: 90, type: "boolean" },
  {
    field: "googleId",
    headerName: "Google ID",
    width: 120,
    valueFormatter: (params) =>
      params.value ? params.value : "No tiene google id",
  },
  { field: "correo", headerName: "Correo", width: 120 },
  { field: "telefono", headerName: "Telefono", width: 120 },
  {
    field: "rol",
    headerName: "Rol",
    valueFormatter: (params) =>
      params.value === 1
        ? "Comprador"
        : params.value === 2
        ? "Sucursal"
        : "Administrador",
    width: 120,
  },
  {
    field: "fechaNacimiento",
    headerName: "Fecha de Nacimiento",
    width: 120,
    valueFormatter: (params) => {
      const fechaFormatted = new Date(params.value);
      return fechaFormatted.toLocaleDateString();
    },
  },
  {
    field: "acciones",
    headerName: "acciones",
    width: 100,
    renderCell: (params) => <BloquearUsuarioActions params={params} />,
  },
];

const BloquearUsuarioActions = ({
  params,
}: {
  params: GridRenderCellParams<any, any, any, GridTreeNodeWithRender>;
}) => {
  const [startBloquearUsuario] = useBloquearUsuarioMutation();
  const [startEliminarUsuario] = useEliminarUsuarioMutation();
  return (
    <>
      {!params.row.bloqueado ? (
        <Tooltip
          title="Bloquear"
          onClick={() => {
            startBloquearUsuario({ correo: params.row.correo })
              .unwrap()
              .then(console.log)
              .catch(console.log);
          }}
        >
          <IconButton>
            <Person />
          </IconButton>
        </Tooltip>
      ) : (
        <Tooltip title="Desbloquear">
          <IconButton>
            <PersonOff />
          </IconButton>
        </Tooltip>
      )}
      <Tooltip title="Eliminar">
        <span>
          <IconButton
            disabled={params.row.eliminado}
            onClick={() => {
              startEliminarUsuario({ correo: params.row.correo })
                .unwrap()
                .then(console.log)
                .catch(console.log);
            }}
          >
            <PersonRemove />
          </IconButton>
        </span>
      </Tooltip>
    </>
  );
};
