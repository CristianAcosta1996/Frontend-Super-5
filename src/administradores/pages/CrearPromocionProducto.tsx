import {
  Snackbar,
  Alert,
  Grid,
  Typography,
  TextField,
  styled,
  Box,
  IconButton,
  Tooltip,
} from "@mui/material";
import { useState, useEffect } from "react";
import { useLocation, useNavigate, useParams } from "react-router-dom";
import { Producto } from "../../interfaces/interfaces";
import { Clear, Check } from "@mui/icons-material";
import { useCrearPromocionProductoMutation } from "../../store/super5/super5Api";

export const CrearPromocionProducto = () => {
  const { id: productoID } = useParams();

  const [fechaInicio, setFechaInicio] = useState<string>();
  const [fechaFin, setFechaFin] = useState<string>();
  const [porcentajeDescuento, setPorcentajeDescuento] = useState<string>();
  const [productoId, setProductoId] = useState<string | undefined>(productoID);

  const [startCrearPromocion] = useCrearPromocionProductoMutation();

  const navigate = useNavigate();
  const [showSnackbar, setShowSnackbar] = useState<{
    show: boolean;
    isError: boolean;
    message: string;
  } | null>(null);

  const handleOnSubmit = (event) => {
    event.preventDefault();
    if (!fechaInicio || !fechaFin || !porcentajeDescuento || !productoId) {
      setShowSnackbar({
        show: true,
        isError: true,
        message: "Todos los campos son obligatorios",
      });
      return;
    }
    const fechaInicioDate = new Date(fechaInicio);
    const fechaFinDate = new Date(fechaFin);

    if (isNaN(Number(porcentajeDescuento)) || fechaInicioDate > fechaFinDate) {
      setShowSnackbar({
        show: true,
        isError: true,
        message: `El porcentaje de descuento debe ser un numero y
           la fecha de inicio de la promocion debe ser anterior a la fecha de fin`,
      });
      return;
    }
    startCrearPromocion({
      fechaDesde: fechaInicio,
      fechaHasta: fechaFin,
      porcDescuentoProducto: +porcentajeDescuento,
      producto: { id: +productoId! },
      tipoPromo: "PRODUCTO",
    })
      .unwrap()
      .then((resp) => {
        console.log(resp);
        setShowSnackbar({
          isError: false,
          show: true,
          message: "La promoción ha sido creada",
        });
      })
      .catch((err) => {
        console.log(err);
        setShowSnackbar({
          show: true,
          isError: true,
          message: "Algo ha salido mal al crear la promoción",
        });
      });
  };
  useEffect(() => {
    if (productoID) return;
    navigate(-1);
  }, [productoID]);

  return (
    <form onSubmit={handleOnSubmit}>
      <Snackbar
        open={showSnackbar?.show || undefined}
        autoHideDuration={2000}
        onClose={() => {
          setShowSnackbar((prev) => ({
            isError: prev?.isError || false,
            message: prev?.message || "",
            show: false,
          }));
        }}
      >
        <Alert
          sx={{ fontSize: 11 }}
          severity={showSnackbar?.isError ? "error" : "success"}
          variant="filled"
        >
          {showSnackbar?.message}
        </Alert>
      </Snackbar>
      <Grid
        className="animate__animated animate__fadeIn animate__faster"
        container
        sx={{
          minHeight: "98vh",
          maxWidth: 800,
          bgcolor: "#333",
          px: 2,
          py: 2,
          color: "#fff",
          margin: "0 auto",
          borderRadius: 1,
        }}
        flexDirection="column"
      >
        <Grid item xs={12}>
          <Typography variant="h5" component="h5">
            Crear promoción
          </Typography>
          <hr />
        </Grid>
        <Grid
          container
          flex={1}
          alignItems="center"
          alignContent="center"
          justifyContent="center"
          gap={3}
        >
          <Grid item xs={12} sm={5} my={1}>
            <CustomTextField
              fullWidth
              label="Fecha de inicio"
              variant="filled"
              size="small"
              type="date"
              InputLabelProps={{ shrink: true }}
              value={fechaInicio || ""}
              onChange={(event) => {
                setFechaInicio(event.target.value);
              }}
            />
          </Grid>
          <Grid item xs={12} sm={5} my={1}>
            <CustomTextField
              fullWidth
              label="Fecha de fin"
              variant="filled"
              size="small"
              type="date"
              InputLabelProps={{ shrink: true }}
              value={fechaFin || ""}
              onChange={(event) => {
                setFechaFin(event.target.value);
              }}
            />
          </Grid>
          <Grid item xs={12} sm={5} my={1}>
            <CustomTextField
              autoComplete="off"
              fullWidth
              label="Porcentaje de descuento"
              variant="filled"
              size="small"
              type="number"
              InputLabelProps={{ shrink: true }}
              value={porcentajeDescuento || ""}
              onChange={(event) => {
                setPorcentajeDescuento(event.target.value);
              }}
            />
          </Grid>
          <Grid item xs={12} sm={5} my={1}>
            <CustomTextField
              disabled
              value={productoId || ""}
              fullWidth
              label="Producto id"
              variant="filled"
              size="small"
              type="text"
              InputLabelProps={{ shrink: true }}
            />
          </Grid>
        </Grid>
        <Grid item xs={12}>
          <Grid
            container
            justifyContent="flex-end"
            gap={2}
            pr={5}
            mt={2}
            alignItems="center"
          >
            <Box>
              <Tooltip title="Cancelar">
                <IconButton
                  onClick={() => {
                    navigate(-1);
                  }}
                  size="small"
                  sx={{
                    border: "1px solid #fff",
                    backgroundImage:
                      "linear-gradient(90deg, rgba(89,41,57,1) 0%, rgba(255,0,86,1) 100%);",
                    "&: hover": {
                      backgroundImage:
                        "linear-gradient(90deg, rgba(89,41,57,1) 0%, rgba(255,0,86,1) 100%);",
                    },
                  }}
                >
                  <Clear sx={{ color: "#fff" }} />
                </IconButton>
              </Tooltip>
            </Box>
            <Box>
              <Tooltip title="Confirmar">
                <IconButton
                  size="small"
                  type="submit"
                  sx={{
                    border: "1px solid #fff",
                    backgroundImage:
                      "linear-gradient(90deg, rgba(89,41,57,1) 0%, rgba(255,0,86,1) 100%);",
                    "&: hover": {
                      backgroundImage:
                        "linear-gradient(90deg, rgba(89,41,57,1) 0%, rgba(255,0,86,1) 100%);",
                    },
                  }}
                >
                  <Check sx={{ color: "#fff" }} />
                </IconButton>
              </Tooltip>
            </Box>
          </Grid>
        </Grid>
      </Grid>
    </form>
  );
};

const CustomTextField = styled(TextField)(() => ({
  "& .MuiFilledInput-root": {
    backgroundColor: "#fff !important",
  },
}));
