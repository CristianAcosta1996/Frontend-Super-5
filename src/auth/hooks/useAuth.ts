import { useAppDispatch } from "../../hooks/hooks";
import {
  useLoginMutation,
  useSignupMutation,
} from "../../store/super5/super5Api";
import {
  startEmailAndPasswordLogin,
  startGoogleSignIn,
  startLogout,
} from "../../store/auth/thunks";

export const useAuth = () => {
  const dispatch = useAppDispatch();

  const [
    startLogin,
    { isLoading: isAuthenticatingLogin, status: statusLogin },
  ] = useLoginMutation();

  const [
    startRegistrarUsuario,
    { isLoading: isAuthenticatingRegistro, status: statusRegistro },
  ] = useSignupMutation();

  const handleLogout = () => {
    dispatch(startLogout());
  };

  const handleLogin = async (email: string, password: string) => {
    const resp: any = await startLogin({
      usuarioOCorreo: email,
      contrasenia: password,
    });
    const token: string = resp.data.token;
    dispatch(startEmailAndPasswordLogin(token));
  };

  const handleGoogleLogin = () => {
    dispatch(startGoogleSignIn());
  };

  const handleRegistrarUsuario = async (
    username: string,
    password: string,
    email: string,
    nombre: string,
    apellido: string,
    phone: string
  ) => {
    const resp = await startRegistrarUsuario({
      nombre,
      correo: email,
      usuario: username,
      contrasenia: password,
      apellido,
      telefono: phone,
      bloqueado: 0,
      eliminado: 0,
      rol: 1,
    });
    console.log(resp);
  };

  return {
    handleRegistrarUsuario,
    handleLogin,
    handleLogout,
    handleGoogleLogin,
    isAuthenticatingLogin,
    statusLogin,
    isAuthenticatingRegistro,
    statusRegistro,
  };
};
