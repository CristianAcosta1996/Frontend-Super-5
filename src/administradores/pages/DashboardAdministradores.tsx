import {
  Fastfood,
  Home,
  Logout,
  Login,
  Person,
  Store,
} from "@mui/icons-material";
import {
  Avatar,
  Box,
  Divider,
  IconButton,
  List,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  SvgIconTypeMap,
  Tooltip,
} from "@mui/material";
import Drawer from "@mui/material/Drawer";
import brandLogo from "../../assets/super5Balnco2.png";
import { useNavigate } from "react-router-dom";

import { OverridableComponent } from "@mui/material/OverridableComponent";
import { useAppSelector } from "../../hooks/hooks";
import { useAuth } from "../../auth/hooks/useAuth";
import { RoutesAdministradores } from "../routes/RoutesAdministradores";

const drawerWidth = 150;
export const DashboardAdministradores = () => {
  const navigate = useNavigate();
  const { status } = useAppSelector((state) => state.auth);
  const { handleLogout } = useAuth();
  return (
    <>
      <Drawer
        variant="permanent"
        anchor="left"
        sx={{
          flexShrink: 0,
          "& .MuiDrawer-paper": {
            width: drawerWidth,
            boxSizing: "border-box",
            backgroundColor: "#333",
            color: "#fff",
          },
        }}
      >
        <Box
          sx={{
            py: 1,
            display: "flex",
            justifyContent: "space-around",
            alignItems: "center",
          }}
        >
          <Tooltip title="Inicio">
            <IconButton
              color="inherit"
              size="small"
              onClick={() => {
                navigate("/administrador");
              }}
            >
              <Home fontSize="small" />
            </IconButton>
          </Tooltip>
          {status === "authenticated" ? (
            <Tooltip
              title="Cerrar Sesion"
              onClick={() => {
                handleLogout();
              }}
            >
              <IconButton color="inherit" size="small">
                <Logout fontSize="small" />
              </IconButton>
            </Tooltip>
          ) : (
            <Tooltip
              title="Iniciar Sesion"
              onClick={() => {
                navigate("/auth/login");
              }}
            >
              <IconButton color="inherit" size="small">
                <Login fontSize="small" />
              </IconButton>
            </Tooltip>
          )}
        </Box>
        <Divider color="#555" />
        <Avatar
          variant="square"
          src={brandLogo}
          sx={{ width: "100%", height: 100, alignSelf: "center" }}
        />
        <Divider color="#555" />
        <List>
          <DashboardListItem
            titulo="Productos"
            handleOnClick={() => {
              navigate("/administrador");
            }}
            Icon={Fastfood}
          />
          <DashboardListItem
            titulo="Usuarios"
            handleOnClick={() => {
              navigate("/administrador/usuarios");
            }}
            Icon={Person}
          />
          <DashboardListItem
            titulo="Sucursal"
            handleOnClick={() => {
              navigate("/administrador/sucursal");
            }}
            Icon={Store}
          />
        </List>
      </Drawer>
      <Box
        sx={{
          width: `calc(100% - ${drawerWidth}px)`,
          ml: `${drawerWidth}px`,
          backgroundColor: "#eee",
          minHeight: "100vh",
          height: "100%",
          overflow: "auto",
          py: 1,
          px: 2,
        }}
      >
        <RoutesAdministradores />
      </Box>
    </>
  );
};

interface DashboardListItem {
  titulo: string;
  handleOnClick: () => void;
  Icon: OverridableComponent<SvgIconTypeMap<object, "svg">> & {
    muiName: string;
  };
}

const DashboardListItem = ({
  Icon,
  handleOnClick,
  titulo,
}: DashboardListItem) => {
  return (
    <ListItemButton onClick={handleOnClick}>
      <ListItemIcon sx={{ color: "inherit", minWidth: 35 }}>
        <Icon />
      </ListItemIcon>
      <ListItemText primaryTypographyProps={{ fontSize: 12 }}>
        {titulo}
      </ListItemText>
    </ListItemButton>
  );
};
