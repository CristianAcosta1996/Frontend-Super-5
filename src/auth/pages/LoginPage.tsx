import { SyntheticEvent, useEffect, useState } from "react";
import {
  Alert,
  Box,
  Button,
  Grid,
  IconButton,
  LinearProgress,
  TextField,
  Typography,
} from "@mui/material";
import { FcGoogle } from "react-icons/fc";

import { FormLayout } from "../layout/FormLayout";
import { useForm } from "../../hooks/useForm";
import brandLogo from "../../assets/super5Balnco2.png";
import { useAuth } from "../hooks/useAuth";
import { useNavigate } from "react-router-dom";

const initialStateForm = {
  username: "",
  password: "",
};

export const LoginPage = () => {
  const {
    handleGoogleLogin,
    handleLogin,
    isAuthenticatingLogin,
    isErrorLogin,
    errorLogin,
    isSuccessLogin,
  } = useAuth();
  const navigate = useNavigate();

  const { username, password, handleInputChange, reset } =
    useForm(initialStateForm);

  const handleFormSubmit = async (event: SyntheticEvent) => {
    event.preventDefault();
    if (!username || !password) return;
    handleLogin(username, password);
    reset();
  };

  return (
    <FormLayout>
      <form onSubmit={handleFormSubmit}>
        <Grid container alignItems="center">
          <Grid container justifyContent="center">
            <img
              src={brandLogo}
              alt="brand logo"
              style={{ height: 100, width: 300, objectFit: "cover" }}
            />
          </Grid>
          <Grid item xs={12} mb={2}>
            <Typography variant="subtitle2" mb={1} color="primary">
              Acceso
            </Typography>
            <TextField
              size="small"
              variant="filled"
              fullWidth
              label="Email/Usuario"
              type="text"
              sx={{ backgroundColor: "#fff", borderRadius: 2 }}
              name="username"
              value={username}
              onChange={handleInputChange}
              disabled={isAuthenticatingLogin}
            />
          </Grid>
          <Grid item xs={12} mb={3}>
            <TextField
              size="small"
              variant="filled"
              fullWidth
              label="Contraseña"
              type="password"
              sx={{ backgroundColor: "#fff", borderRadius: 2 }}
              name="password"
              value={password}
              onChange={handleInputChange}
              disabled={isAuthenticatingLogin}
            />
          </Grid>
          <Grid container mb={1} justifyContent="space-between">
            <Grid item xs={12}>
              <LoginButton
                titulo="Iniciar sesion"
                type="submit"
                disabled={isAuthenticatingLogin}
              />
            </Grid>

            <Grid container justifyContent="center" alignItems="center">
              <Typography color="#fff" sx={{ fontSize: 12 }}>
                ¿No tienes una cuenta?
              </Typography>
              <Button
                variant="text"
                color="primary"
                sx={{ textTransform: "capitalize", fontSize: 12 }}
                disabled={isAuthenticatingLogin}
                onClick={() => {
                  navigate("/auth/signup");
                }}
              >
                Regístrate.
              </Button>
            </Grid>
            <Grid container justifyContent="center" alignItems="center">
              <Typography color="#fff" sx={{ fontSize: 12 }}>
                ¿Olvidaste tu contraseña?
              </Typography>
              <Button
                size="small"
                variant="text"
                sx={{
                  textTransform: "capitalize",
                  fontSize: 12,
                  textDecoration: "underline",
                  color: "#fff",
                }}
                disabled={isAuthenticatingLogin}
                onClick={() => navigate("/auth/recuperarcontrasena")}
              >
                recuperar contraseña
              </Button>
            </Grid>
          </Grid>
          <Grid item xs={12} mb={1} sx={{ position: "relative" }}>
            <SocialMediaLoginLabel />
            <Grid container justifyContent="center">
              <IconButton
                onClick={() => handleGoogleLogin()}
                sx={{
                  backgroundColor: "#fff",
                  color: "rgb(0,130,255)",
                  "&:hover": { backgroundColor: "#ddd" },
                }}
              >
                <FcGoogle />
              </IconButton>
            </Grid>
          </Grid>
        </Grid>
      </form>
      {isErrorLogin && (
        <Alert
          variant="filled"
          severity="error"
          className="animate__animated animate__fadeIn"
        >
          Email/Usuario o Contraseña invalidos
        </Alert>
      )}
      {isSuccessLogin && (
        <Alert
          variant="filled"
          severity="success"
          className="animate__animated animate__fadeIn"
        >
          Login exitoso!.
        </Alert>
      )}
      {isAuthenticatingLogin && (
        <Box
          sx={{
            width: "100%",
            position: "absolute",
            top: "100%",
            left: "50%",
            transform: "translate(-50%, -100%)",
          }}
        >
          <LinearProgress />
        </Box>
      )}
    </FormLayout>
  );
};

const SocialMediaLoginLabel = () => (
  <Box
    sx={{
      position: "relative",
      "&::after": {
        content: '""',
        display: "block",
        width: "100%",
        position: "absolute",
        top: "50%",
        borderBottom: "2px solid #fff",
        zIndex: 0,
      },
      mb: 2,
    }}
  >
    <Typography
      variant="subtitle2"
      component="span"
      color="#fff"
      bgcolor="#333"
      textAlign="center"
      sx={{
        position: "relative",
        display: "inline-block",
        zIndex: 1,
        top: 0,
        left: "50%",
        transform: "translateX(-50%);",
        px: 2,
      }}
    >
      O inicia sesion con
    </Typography>
  </Box>
);

interface LoginButtonProps {
  titulo: string;
  handleOnClick?: () => void;
  type?: "submit" | "button" | "reset";
  disabled: boolean;
}
const LoginButton = ({
  titulo,
  handleOnClick,
  type,
  disabled,
}: LoginButtonProps) => (
  <Button
    size="small"
    fullWidth
    variant="contained"
    sx={{ mb: 1 }}
    onClick={handleOnClick}
    type={type}
    disabled={disabled}
  >
    {titulo}
  </Button>
);
