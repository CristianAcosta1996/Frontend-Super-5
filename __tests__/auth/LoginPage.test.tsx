import { render, screen, fireEvent } from "@testing-library/react";
import { LoginPage } from "../../src/auth/pages/LoginPage";
import React from "react";

import { useAuth } from "../../src/auth/hooks/useAuth";

jest.mock("react-router-dom");
jest.mock("../../src/auth/hooks/useAuth");
jest.mock(
  "../../src/assets/super5Balnco2.png",
  () => "../../src/assets/super5Balnco2.png"
);

describe("Pruebas en el componente <LoginPage/>", () => {
  test("Loguear un usuario correctamente", () => {
    const handleLoginMock = jest.fn();

    (useAuth as jest.Mock).mockReturnValue({
      handleGoogleLogin: jest.fn(),
      handleLogin: handleLoginMock,
      isAuthenticatingLogin: false,
      isErrorLogin: false,
      isSuccessLogin: false,
    });

    render(<LoginPage />);

    const correo = "test@example.com";
    const password = "password";
    const usernameInput = screen.getByLabelText("Email/Usuario");
    const passwordInput = screen.getByLabelText("ContraseÃ±a");
    const loginButton = screen.getByRole("button", { name: "Iniciar sesion" });

    fireEvent.change(usernameInput, { target: { value: correo } });
    fireEvent.change(passwordInput, { target: { value: password } });

    fireEvent.click(loginButton);

    expect(handleLoginMock).toBeCalled();
    expect(handleLoginMock).toBeCalledTimes(1);
    expect(handleLoginMock).toBeCalledWith(correo, password);
  });

  test("Mostrar mensaje de usuario logueado correctamente", () => {
    const handleLoginMock = jest.fn();
    (useAuth as jest.Mock).mockReturnValue({
      handleGoogleLogin: jest.fn(),
      handleLogin: handleLoginMock,
      isAuthenticatingLogin: false,
      isErrorLogin: false,
      isSuccessLogin: true,
    });

    render(<LoginPage />);

    const successfulLoginMessage = screen.getByLabelText(
      "successful-login-message"
    );
    expect(successfulLoginMessage).toBeTruthy();
    expect(successfulLoginMessage.textContent).toBe(
      "Usuario logueado correctamente!"
    );
  });

  test("comparar <LoginPage /> con su snapshot", () => {
    const { container } = render(<LoginPage />);
    expect(container).toMatchSnapshot();
  });
});
