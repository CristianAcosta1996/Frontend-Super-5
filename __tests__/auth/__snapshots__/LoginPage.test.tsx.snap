// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <LoginPage/> comparar <LoginPage /> con su snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1uygidu"
  >
    <div
      class="MuiBox-root css-8suzv6"
    >
      <form>
        <div
          class="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
          >
            <img
              alt="brand logo"
              src="../../src/assets/super5Balnco2.png"
              style="height: 100px; width: 300px; object-fit: cover;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-yp0tso-MuiGrid-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-1ovtx3w-MuiTypography-root"
            >
              Acceso
            </h6>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1kpu9k-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r4:"
                id=":r4:-label"
              >
                Email/Usuario
              </label>
              <div
                class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-2y464i-MuiInputBase-root-MuiFilledInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                  id=":r4:"
                  name="username"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-5pv5pv-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1kpu9k-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r5:"
                id=":r5:-label"
              >
                Contraseña
              </label>
              <div
                class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-2y464i-MuiInputBase-root-MuiFilledInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                  id=":r5:"
                  name="password"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container css-14usrz-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth css-1c5ch3b-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Iniciar sesion
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-1hbmzt3-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-h3yix-MuiTypography-root"
              >
                ¿No tienes una cuenta?
              </p>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-9479yy-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Regístrate.
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-1hbmzt3-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-h3yix-MuiTypography-root"
              >
                ¿Olvidaste tu contraseña?
              </p>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-a7ydnc-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                recuperar contraseña
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-rv5gml-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-1muppml"
            >
              <span
                class="MuiTypography-root MuiTypography-subtitle2 css-492r7h-MuiTypography-root"
              >
                O inicia sesion con
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1kryon7-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  enable-background="new 0 0 48 48"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1.1"
                  viewBox="0 0 48 48"
                  width="1em"
                  x="0px"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0px"
                >
                  <path
                    d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
                    fill="#FFC107"
                  />
                  <path
                    d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
                    fill="#FF3D00"
                  />
                  <path
                    d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
                    fill="#4CAF50"
                  />
                  <path
                    d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
                    fill="#1976D2"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
      <div
        aria-label="successful-login-message"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-filledSuccess MuiAlert-filled animate__animated animate__fadeIn css-camg44-MuiPaper-root-MuiAlert-root"
        role="alert"
      >
        <div
          class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="SuccessOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
        >
          Usuario logueado correctamente!
        </div>
      </div>
    </div>
  </div>
</div>
`;
